openedx/core/djangoapps/programs/utils.py

```diff
@@ -334,6 +335,14 @@ def _attach_course_run_upgrade_url(self, run_mode):
         else:
             run_mode['upgrade_url'] = None
 
+    def _attach_course_run_can_enroll(self, run_mode):
+        run_mode['can_enroll'] = bool(has_access(self.user, 'enroll', self.course_overview))
+
+    def _attach_course_run_registered(self, run_mode):
+        # avoiding cyclic import error
+        from lms.djangoapps.courseware.views.views import registered_for_course
+        run_mode['registered'] = registered_for_course(self.course_overview, self.user)
```

Comment:
I think we already get this with the `is_enrolled` flag added [above](https://github.com/edx/edx-platform/pull/14499/files#diff-1072896d9c36e00edfdd4998d9e13d22R303).